partner_validate_vat/models/res_partner.py

```diff
@@ -0,0 +1,35 @@
+# coding: utf-8
+# Copyright 2017 Vauxoo (https://www.vauxoo.com) <info@vauxoo.com>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from openerp import models, api, fields, _
+from openerp.exceptions import ValidationError
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    vat = fields.Char(copy=False)
+
+    @api.constrains('vat', 'country_id', 'commercial_partner_id')
```

Comment:
Check if here you won't have a error with faster creation:
 - https://github.com/odoo/odoo/pull/18053